### ISSUE TITLE
Cant download Result when logged in Using OKTA SSO

### ISSUE THREAD

**Issue by jietnitish** [SitecorePowerShell/Console#1236]
### Expected Behavior

User should be able to download the report in CSV and/or Excel

### Actual Behavior

Nothing is happening when you click CSV or Excel Button when logged in via OKTA SSO 
### Steps to Reproduce the Problem

Sitecore Version 9.3
Powershell Module Version 6

1. Login to CMS via SSO (Azure / OKTA)
2. Navigate to Start --> Reporting --> PS Reports-->Content Audit-->Broken Link
3. Execute th script and make sure you get some output
4. Click on CSV or Excel

Validate the same behavior using normal Sitecore Login and this will work.
When I checked the network tab for all the call, I can see one differene in working scenario vs non working scenario

Request : /sitecore/shell/applications/powershell/powershelllistview?sid=9510dfc9-e66f-45ba-8178-64c330b4fc42 

Response for Working Scenario
{
"commands": [
{
"command": "UnregisterKeyGroup",
"value": "PowerShellRibbon"
},
{
"command": "SetInnerHtml",
"value": "SOME DATA GOES HERE...REMOVED FOR TICKET",
"id": "RibbonPanel"
},
**{
"command": "SetPipeline",
"value": "98715B6C8CA74550B0D8E51DB94F633A"
},
{
"command": "Eval",
"value": "spe.DownloadReport('A14F173DE53C4B218A76DEBD8F7CF1CD');"
}**
]
}

Response for Non Working Scenario

{
"commands": [
{
"command": "UnregisterKeyGroup",
"value": "PowerShellRibbon"
},
{
"command": "SetInnerHtml",
"value": "SOME DATA GOES HERE...REMOVED FOR TICKET",
"id": "RibbonPanel"
}
]
}

If you see, in non working scenario, 2 commands are not coming in response which might be causing this issue,

**Comment by michaellwest** [comment]
@jietnitish Are there other issues you have noticed that do not work? Do you have the `Spe.IdentityServer.config` enabled?

**Comment by jietnitish** [comment]
Hi @michaellwest , Yes... above config file is enabled
Also if I try to execute any script from Powershell ISE, then I am getting a popup to Elevate Session, which I am not able to do so because I have logged in via OKTA SSO and we are not creating persisten user.
If I go via Start --> Reporting --> PS Reports-->Content Audit-->Broken Link route, then I am able to execute the script, but not able to download the report. 

As in my above comment, some response is missing in JSON which could be causing this issue.

**Comment by michaellwest** [comment]
In SPE 6.2 you can change the elevate access to use "Confirm" instead of "Password". That could get you past the dialogs asking for password. 

If you can apply the shields down patch to Allow everything instead of requiring the password then we can narrow down which option is giving you trouble.

**Comment by jietnitish** [comment]
@michaellwest I did not understand "If you can apply the shields down patch to Allow everything instead of requiring the password then we can narrow down which option is giving you trouble."
Also we are using SPE 6.0

**Comment by jietnitish** [comment]
@michaellwest  Is it related to this https://alan-null.github.io/2017/01/spe-dev-config ?

**Comment by michaellwest** [comment]
Yes that's what I was thinking about. In the patch there are sections for each of the types of services offered by SPE. The ability to download is probably the handle service.. My guess is it fails to authenticate when using that feature.

**Comment by jietnitish** [comment]
@michaellwest , using the patch resolved the "Elevate Session" problem. But still I am unable to downaload the result.

**Comment by jietnitish** [comment]
When trying download option via SSO Login (OKTA)
![SSO1](https://user-images.githubusercontent.com/82812182/115642802-f655bf00-a35e-11eb-97ed-c8d2371b0146.png)

When trying download option via Sitecore Login
![SSO2](https://user-images.githubusercontent.com/82812182/115642845-053c7180-a35f-11eb-88ed-8f1d4e8c482e.png)

Network Tab Response for SSO Call (Where download is not working)
![SSO3](https://user-images.githubusercontent.com/82812182/115642885-14bbba80-a35f-11eb-8961-63e657e1db74.png)

Network Tab Response for Normal Login Call (Where download is working)
![SSO4](https://user-images.githubusercontent.com/82812182/115642910-23a26d00-a35f-11eb-91ec-95f64a140338.png)

**Comment by michaellwest** [comment]
Taking a guess, the user is not detected in one of these:

```powershell
<handleDownload enabled="true" requireSecureConnection="false">
          <authorization>
            <add Permission="Allow" IdentityType="Role" Identity="sitecore\PowerShell Extensions Remoting" />
            <add Permission="Allow" IdentityType="Role" Identity="sitecore\Sitecore Client Users" />
            <add Permission="Allow" IdentityType="Role" Identity="sitecore\IsAdministrator" />
          </authorization>
</handleDownload>
```

**Comment by jietnitish** [comment]
We have looked at this option as well. User logged in via OKTA is Admin User. I can try adding a role here and map the same with logged in OKTA User

**Comment by michaellwest** [comment]
Do you think this fixes your issue? #1227

**Comment by jietnitish** [comment]
Hi @michaellwest , I allowed a role handleDownload authorization, but still issue exist.
For testingpurpose. I explictly added a role "sitecore\IsAdministrator" to incoming user while mapping cliams to Sitecore role, but that also does not resolve the issue.

**Comment by jietnitish** [comment]
@michaellwest  can you share the package mentioned at 1227 ?

**Comment by michaellwest** [comment]
I never created it as the user was able to build their own. Essentially compile the latest version of SPE, or branch of the version you have, and then add in the fix.

**Comment by jietnitish** [comment]
Hey @michaellwest , tried above by customizing the code, still same issue

**Comment by michaellwest** [comment]
Crazy idea, but what if you change the "IdentityType" to "User" and then put a wildcard `*`. Curious if that will match all users.

**Comment by jietnitish** [comment]
I will try and let you know

**Comment by jietnitish** [comment]
Unfortunately, this is also not working. Tried with * and username both, but no success.

**Comment by AdamNaj** [comment]
With SSO login using Identity server you most likely need to switch on the Owin compatibility mode in SPE. 
The file is included with SPE 7.0 but wasn't there earlier or was shipped in a disabled state.
You can find the file [here](https://github.com/SitecorePowerShell/Console/blob/master/src/Spe/App_Config/Include/Spe/Spe.IdentityServer.config)

In the meantime closing this issue as stale. If anyone finds this problem still relevant and the above does not solve the problem - please re-open the issue.
